bin/cache_repo

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+AWS_BUCKET=$1
+
+DATE=$(date +"%Y-%m-%d")
```

Comment:
Apologies if I missed where this was already discussed. But could you elaborate on the rationale of caching the the repo on a daily basis?

I noticed that the scheduled pipeline is [part of the WordPress iOS repo](https://github.com/wordpress-mobile/WordPress-iOS/pull/16935/files#diff-45d463ed168fd7f9dae18185bf9327f3b489aa63a91c0798a5bbeb5837f9d15f). Have you considered moving it to its own repository to centralize the schedule configuration? Or is the idea behind it that every repo has different caching needs, so it makes sense for each to setup its own caching "cron" job?

Reply:
> Have you considered moving it to its own repository to centralize the schedule configuration? 

Good question! The main reason for this is the fact that each project has a different SSH key, so having one project that does all of the cloning would be kinda complex, whereas we can set-it-and-forget it this way.

> could you elaborate on the rationale of caching the the repo on a daily basis

For sure – we could do this on basically any basis, doesn't really matter – the main idea is that for large repos we don't need to fetch the whole thing from GitHub every time – instead we can use the cached repo as a [reference repo](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---reference-if-ableltrepositorygt) when cloning, which is significantly faster.

Reply:
> we could do this on basically any basis, doesn't really matter – the main idea is that for large repos we don't need to fetch the whole thing from GitHub every time – instead we can use the cached repo as a reference repo when cloning, which is significantly faster – the only thing downloaded from GitHub is differences since the last cache.

This is super neat! 🤯 

Are the two rationales, for why using a local schedule pipeline and for the repo caching necessity, documented somewhere?

Finally, regarding the centralization, I guess by having the logic live in the plugin here, we centralize the core of the work: defining the logic. Each project can then opt-in and configure the caching schedule as it prefers. _**If**_ we'll need to, we could make it more flexible by allowing consumers to set their own key for the caching.

Reply:
> we could make it more flexible by allowing consumers to set their own key for the caching

Certainly, but I figured we'd go for convention over configuration here – if both sides use the Buildkite pipeline slug then it "just works" – hopefully nobody ever has reason to change it 😅